mux/src/domain.rs

```diff
@@ -119,6 +119,18 @@ pub trait Domain: Downcast + Send + Sync {
                     anyhow::anyhow!("pane {} not found in its containing tab!?", src_pane_id)
                 })?;
 
+                // May need to update pane_index if src_pane was also in the same tab
+                if src_tab.tab_id() == tab.tab_id() {
+                    pane_index = match tab
+                        .iter_panes_ignoring_zoom()
+                        .iter()
+                        .find(|p| p.pane.pane_id() == pane_id)
+                    {
+                        Some(p) => p.index,
+                        None => anyhow::bail!("invalid pane id {}", pane_id),
+                    };
+                }
+
```

Comment:
Let's make this unconditional but move it to just before the `tab.split_and_insert` call on line 142:

```suggestion
                   // May need to update pane_index if the logic above mutated the tab contents
                   let pane_index = match tab
                        .iter_panes_ignoring_zoom()
                        .iter()
                        .find(|p| p.pane.pane_id() == pane_id)
                    {
                        Some(p) => p.index,
                        None => anyhow::bail!("invalid pane id {}", pane_id),
                    };
```